iib/workers/tasks/utils.py

```diff
@@ -723,6 +723,13 @@ def run_cmd(cmd, params=None, exc_msg=None, strict=True):
     if strict and response.returncode != 0:
         log.error('The command "%s" failed with: %s', ' '.join(cmd), response.stderr)
         if cmd[0] == 'opm':
+            # The new OPM shows the error at the beginning, let's try with this first
```

Comment:
I would suggest to use `opm` version here in the comment. "new OMP" is relative and it might mean something else in few years.

Reply:
As far as I looked on OPM versions each one puts the error on top, I don't know why they decided to use the reverse search but the comment `#Capture the error message right before the help display` might indicate that it was always on top.

So I believe I was touching in the wrong part of the code. The regex is not broken, if we look in the [logs](https://iib.qe.engineering.redhat.com/api/v1/builds/5924/logs) near bottom it says:

```iib.exceptions.IIBError: Failed to deprecate the bundles: error deprecating bundle registry-proxy.engineering.redhat.com/rh-osbs/amq7-amq-streams-rhel7-operator-metadata@sha256:25b8153b1f5cace8963715dad8019b07fbbc2501d42fb21b189193baa596a41d: Bundle deprecation causing default channel removal```

Which indicates IIB raised the exception, thus regex always worked fine.

Doing the tests locally I confirmed that original regex works fine, so that's not the problem, the problem is that user received the message `An unknown error occurred. See logs for details` which is the default `exec_msg` when not set. Instead, the user might receive  `Bundle deprecation causing default channel removal` or something similar.

Thus the solution need to be different.

As I see [here](https://github.com/operator-framework/operator-registry/releases/tag/v1.18.0) OPM 1.18 seems to have support for default channel removal, so I need to implement something to detect if OPM returned `Bundle deprecation causing default channel removal` and then return this message to the user instead of default `exec_msg`

For the moment I'll close the PR and put this information on JIRA.

Thanks for reviewing!